codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/OperationGenerator.java

```diff
@@ -141,116 +126,98 @@ public void run() {
     }
 
     /**
      * Generate operation protocol middleware helper.
      */
-    private void generateOperationProtocolMiddlewareHelper() {
+    private void generateOperationProtocolMiddlewareAdders() {
         if (protocolGenerator == null) {
             return;
         }
-        writer.openBlock("func $L (stack *middleware.Stack) {", "}",
-                getOperationProtocolMiddlewareHelperName(operation), () -> {
-
-                    // Add request serializer middleware
-                    String serializerMiddlewareName = ProtocolGenerator.getSerializeMiddlewareName(
-                            operation.getId(),
-                            protocolGenerator.getProtocolName());
-                    writer.write("stack.Serialize.Add(&$L{}, middleware.After)", serializerMiddlewareName);
+        writer.addUseImports(SmithyGoDependency.SMITHY_MIDDLEWARE);
 
-                    // Adds response deserializer middleware
-                    String deserializerMiddlewareName = ProtocolGenerator.getDeserializeMiddlewareName(
-                            operation.getId(),
-                            protocolGenerator.getProtocolName());
-                    writer.write("stack.Deserialize.Add(&$L{}, middleware.After)", deserializerMiddlewareName);
-                    writer.addUseImports(SmithyGoDependency.SMITHY_MIDDLEWARE);
-                });
+        // Add request serializer middleware
+        String serializerMiddlewareName = ProtocolGenerator.getSerializeMiddlewareName(
+                operation.getId(),
+                protocolGenerator.getProtocolName());
+        writer.write("err = stack.Serialize.Add(&$L{}, middleware.After)", serializerMiddlewareName);
+        writer.write("if err != nil { return err }");
+
+        // Adds response deserializer middleware
+        String deserializerMiddlewareName = ProtocolGenerator.getDeserializeMiddlewareName(
+                operation.getId(),
+                protocolGenerator.getProtocolName());
+        writer.write("err = stack.Deserialize.Add(&$L{}, middleware.After)", deserializerMiddlewareName);
+        writer.write("if err != nil { return err }");
     }
 
     /**
-     * Get middleware helper name for Protocol middleware.
+     * Returns the name of the operation's middleware mutator function, that adds all middleware for the operation to
+     * the stack.
      *
-     * @param operation Operation for which middleware helpers are generated.
-     * @return protocol middleware helper name.
+     * @param operation symbol for operation
+     * @return name of function
      */
-    private String getOperationProtocolMiddlewareHelperName(OperationShape operation) {
-        return String.format("add%s_serdeOp%sMiddlewares",
-                protocolGenerator.getProtocolName(), operation.getId().getName());
+    public static String getAddOperationMiddlewareFuncName(Symbol operation) {
+        return String.format("addOperation%sMiddleware", operation.getName());
```

Comment:
Nit: If you are gonna regenerate clients, consider changing this to be plural (Middleware/s)